File: clients/macos/SyncBridgeMac/Core/Network/FileRouting.swift
// FileRouting.swift — automatic file transfer routing (clipboard uses relay separately).

import Foundation

enum FileRouting {
    static let relayMaxBytes = 100 * 1024 * 1024

    static func resolve(
        fileSizeBytes: Int,
        fileCount: Int = 1,
        isFolder: Bool = false
    ) -> (transferMode: String, route: String, fallback: String?) {
        let attemptWebRtc = fileSizeBytes > relayMaxBytes || fileCount > 1 || isFolder
        if !attemptWebRtc {
            return ("relay", "cloud", nil)
        }
        return (
            "webrtc",
            "webrtc",
            "WebRTC attempted — automatic relay fallback if P2P unavailable"
        )
    }
}

File: clients/android/app/src/main/java/com/syncbridge/android/network/FileRouting.kt
package com.syncbridge.android.network

/** Automatic file routing — clipboard always uses relay separately. */
object FileRouting {
    const val RELAY_MAX_BYTES = 100L * 1024 * 1024

    data class Route(
        val transferMode: String,
        val route: TransferRoute,
        val fallbackReason: String? = null,
    )

    fun resolve(
        fileSizeBytes: Long,
        fileCount: Int = 1,
        isFolder: Boolean = false,
    ): Route {
        val attemptWebRtc = fileSizeBytes > RELAY_MAX_BYTES || fileCount > 1 || isFolder
        if (!attemptWebRtc) {
            return Route("relay", TransferRoute.Cloud)
        }
        return Route(
            transferMode = "webrtc",
            route = TransferRoute.WebRtc,
            fallbackReason = "WebRTC attempted — automatic relay fallback if P2P unavailable",
        )
    }
}
